apiserver/facades/client/spaces/spaces_test.go

```diff
@@ -900,45 +907,25 @@ func (s *LegacySuite) TestShowSpaceError(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, "getting environ: retrieving model config: boom")
 }
 
-func (s *LegacySuite) TestCreateSpacesModelConfigError(c *gc.C) {
-	defer s.setupMocks(c).Finish()
-	s.makeAPI(c)
-
-	apiservertesting.SharedStub.SetErrors(
-		errors.New("boom"), // Backing.ModelConfig()
-	)
-
-	args := params.CreateSpacesParams{}
-	_, err := s.facade.CreateSpaces(stdcontext.Background(), args)
-	c.Assert(err, gc.ErrorMatches, "getting environ: retrieving model config: boom")
-}
-
 func (s *LegacySuite) TestCreateSpacesProviderOpenError(c *gc.C) {
 	defer s.setupMocks(c).Finish()
-	s.makeAPI(c)
 
-	apiservertesting.SharedStub.SetErrors(
-		nil,                // Backing.ModelConfig()
-		nil,                // Backing.CloudSpec()
-		errors.New("boom"), // Provider.Open()
-	)
+	s.providerGetter = func(stdcontext.Context) (environs.NetworkingEnviron, error) {
+		return nil, errors.New("boom")
+	}
+	s.makeAPI(c)
 
 	args := params.CreateSpacesParams{}
 	_, err := s.facade.CreateSpaces(stdcontext.Background(), args)
 	c.Assert(err, gc.ErrorMatches,
-		`getting environ: creating environ for model \"stub-zoned-networking-environ\" \(.*\): boom`)
+		`getting environ: boom`)
 }
 
 func (s *LegacySuite) TestCreateSpacesNotSupportedError(c *gc.C) {
```

Comment:
There should also be a test here to assert what happens when GetProvider returns not supported as per comments above.

Reply:
I changed TestCreateSpacesGetProviderError to do this